src/dune_engine/diff_promotion.mli

```diff
@@ -45,3 +45,4 @@ type files_to_promote =
 
 val promote_files_registered_in_last_run : files_to_promote -> unit
 val display : files_to_promote -> unit Fiber.t
```

Comment:
@snowleopard is `Diff_promotion.display` used internally? If not, I can move it to the CLI.

Reply:
I think it's unused, so, yes let's move it. Thanks!

Reply:
Done in https://github.com/ocaml/dune/pull/9931